### PR TITLE
Add Obligations to fragile-goods and supply-agreement-perishable-goods

### PR DESCRIPTION
**Pull request by mttrbrts**
Add obligation emission to:
- fragile-goods
- supply-agreement-perishable-goods

Fix test failure in: 
- perishable-goods

Make stateful:
- one-time-payment

### CODE REVIEW

File: src/fragile-goods/lib/logic.ergo [hunk 1]
```diff
@@ -39,19 +39,36 @@ contract FragileGoods over FragileGoodsClause {
...
-						currencyCode: currency
-					}
-				}
+        if (duration) > contract.deliveryLimitDuration.amount then
+          define variable monetaryAmount = new org.accordproject.money.MonetaryAmount {
+            doubleValue: amount - contract.lateDeliveryPenalty.doubleValue,
+            currencyCode: currency
+          };
+          emit new org.accordproject.cicero.runtime.PaymentObligation {
+            contract: "NOT_SET",
```
Comment (mttrbrts) [inline src/fragile-goods/lib/logic.ergo]: @jeromesimeon Note the hard-coded contract ID here. This is required as this is a clause template, but we don't yet allow independent declarations of clauses without contracts.

File: src/fragile-goods/lib/logic.ergo [hunk 2]
```diff
@@ -15,7 +15,7 @@
 namespace io.clause.demo.fragileGoods
 
 contract FragileGoods over FragileGoodsClause {
-  clause fragilegoods(request : DeliveryUpdate) : PayOut {
+  clause fragilegoods(request : DeliveryUpdate) : PayOut emits org.accordproject.cicero.runtime.PaymentObligation {
```
Comment (dselman) [inline src/fragile-goods/lib/logic.ergo]: Do we support `import` so that you don't have to use a FQN?

File: src/one-time-payment/lib/logic.ergo [hunk 3]
```diff
@@ -14,16 +14,42 @@
...
-    return new InitPaymentResponse{}
+    return new InitResponse{}
   }
+
+  /**
+	 * Update the contract state once we've recieved a payment
+	 */
+  clause paymentReceived(request : PaymentReceivedRequest): PaymentReceivedResponse {
+    enforce (state.status = "OBLIGATION_EMITTED") 
+    else throw "Either a payment obligation hasn't yet been emitted by the contract or payment notification has already been received";
```
Comment (dselman) [inline src/one-time-payment/lib/logic.ergo]: We should add an issue to support globalization/localization of strings like this.

File: src/fragile-goods/lib/logic.ergo (hunk 2 above)
Comment (mttrbrts) [reply src/fragile-goods/lib/logic.ergo]: Yes, this is in Ergo 0.1.1. I'll update the template once we get that into Cicero

File: src/one-time-payment/lib/logic.ergo (hunk 3 above)
Comment (mttrbrts) [reply src/one-time-payment/lib/logic.ergo]: https://github.com/accordproject/ergo/issues/269

**Pull request merged**